pkgs/development/python-modules/torch/default.nix

```diff
@@ -65,10 +65,10 @@ assert !cudaSupport || magma.cudaPackages.cudatoolkit == cudatoolkit;
 let
   setBool = v: if v then "1" else "0";
 
-  # https://github.com/pytorch/pytorch/blob/v1.13.1/torch/utils/cpp_extension.py#L1751
+  # https://github.com/pytorch/pytorch/blob/v2.0.1/torch/utils/cpp_extension.py#L1744
   supportedTorchCudaCapabilities =
     let
-      real = ["3.5" "3.7" "5.0" "5.2" "5.3" "6.0" "6.1" "6.2" "7.0" "7.2" "7.5" "8.0" "8.6"];
+      real = ["3.5" "3.7" "5.0" "5.2" "5.3" "6.0" "6.1" "6.2" "7.0" "7.2" "7.5" "8.0" "8.6" "8.9" "9.0"];
```

Comment:
Isn't it time to remove this and just use the global `cudaFlags` as defaults?

Reply:
Oh right, I forget, we do use defaults from `cudaFlags`, it's a question of whether to respect upstream's `supported_arches` list (whose purpose is unclear to me)

Reply:
AFAIK PyTorch has that there to guard against scenarios where people use extensions compiled with a version of CUDA or targeting a capability different than PyTorch.

I’m sure there are other reasons, but if it really comes down to being a runtime check to ensure extensions target the same capabilities, we (Nixpkgs) can definitely do without that check.

There is work in the future to patch it out so PyTorch respects whatever we ask it to build, but I wanted something quick to merge.